sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -187,6 +187,64 @@ case class Like(left: Expression, right: Expression)
   override def matches(regex: Pattern, str: String): Boolean = regex.matcher(str).matches()
 
   override def toString: String = s"$left LIKE $right"
+
+  override protected def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    val patternClass = classOf[Pattern].getName
+
+    val literalRight: String = right match {
+      case x @ Literal(value: String, StringType) => escape(value)
+      case _ => null
+    }
+
+    val leftGen = left.gen(ctx)
+    val rightGen = right.gen(ctx)
+
+    val patternCode =
+      if (literalRight != null) {
+        s"${patternClass} pattern = $patternClass.compile($literalRight);"
+      } else {
+        s"""
+          StringBuilder regex = new StringBuilder("(?s)");
+          for (int idx = 1; idx < rightStr.length(); idx++) {
+            char prev = rightStr.charAt(idx - 1);
+            char curr = rightStr.charAt(idx);
+            if (prev == '\\\\') {
+              if (curr == '_') {
+                regex.append("_");
+              } else if (curr == '%') {
+                regex.append("%");
+              } else {
+                regex.append(${patternClass}.quote("\\\\" + curr));
+              }
+            } else {
+              if (curr != '\\\\') {
+                if (curr == '_') {
+                  regex.append(".");
+                } else if (curr == '%') {
+                  regex.append(".*");
+                } else {
+                  regex.append(${patternClass}.quote((new Character(curr)).toString()));
+                }
+              }
+            }
+          }
+          ${patternClass} pattern = ${patternClass}.compile(regex.toString());
+        """
+      }
```

Comment:
I am not sure I get your point. I think we don't want to call `escape` in the generated java codes?